cmd/fetch-tsa-certs/fetch_tsa_certs.go

```diff
@@ -79,18 +93,18 @@ var (
 	outputPath     = flag.String("output", "", "Path to the output file")
 )
 
-func fetchCertificateChain(ctx context.Context, parent, intermediateKMSKey, leafKMSKey, tinkKeysetPath, tinkKmsKey string,
+func fetchCertificateChain(ctx context.Context, root, parentKMSKey, leafKMSKey, tinkKeysetPath, tinkKmsKey string,
 	client *privateca.CertificateAuthorityClient) ([]*x509.Certificate, error) {
-	intermediateKMSSigner, err := kms.Get(ctx, intermediateKMSKey, crypto.SHA256)
+	parentKMSSigner, err := kms.Get(ctx, parentKMSKey, crypto.SHA256)
```

Comment:
Can we make the hash configurable via a flag? I believe this will err out if the hash func doesn't match what the KMS provider supports, e.g. ecdsa-p384 needs sha-384. We can default to sha256.

Reply:
I added a few options. The method call is a bit unwieldy now (with so many options) but I resisted urge to do more cleanup in the same PR

Reply:
Scratch that: I tried to do this by just providing other values to `kms.Get()` and the signature algorithm in the certificate does not change... 
* This makes sense when you consider that the (GCP) KMS key has the hash function baked in
* I'm not quite sure why `kms.Get()` requires a hash function as an argument?

In any case implementing this may not be the 15 min job I imagined: I would rather not include this in the PR. Let's file an issue if the feature is needed.

Reply:
Sigh, sorry for making you go down this rabbit hole. Only the [hashivault](https://github.com/sigstore/sigstore/blob/0e0ac2ead2b72d0bb713eaa126778706179b3d8a/pkg/signature/kms/hashivault/client.go#L43) KMS provider uses the hash. I thought all did to determine which hash to use when computing the digest to sign...oh well. Crypto agility is fun.